### PR TITLE
Update manual for Smart Proxy puppetca and the new providers

### PR DESCRIPTION
**Pull request by dgoetz**
Update of the documentation was requested by @tbrisker in https://github.com/theforeman/smart-proxy/pull/586. @timogoebel can you please review?

Should the changes also be backported to older manuals? If yes, 1.19 included the change to make puppetca plug-able and 1.20 added the token based mechanism, correct?

### CODE REVIEW

**Comment by tbrisker** [comment]
Thanks for the PR @dgoetz ! Would be good to update the manuals for 1.19 and 1.20 as well since they are still supported.

**Comment by dgoetz** [comment]
Ok, I wait for a review on the content and then backport to the older ones. So please do not merge before.

File: _includes/manuals/nightly/4.3.7_smartproxy_puppetca.md [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+This will create an autosign entry for a host during deployment and remove it when deployment is finished.
+Furthermore it allows you to manage entries manually using the Foreman WebUI.
+
+The **autosignfile** setting in `puppetca_hostname_whitelisting.yml` is used to find autosign.conf:
 
 <pre>
 :autosignfile: /etc/puppet/autosign.conf
 </pre>
 
+The location of the file you can determine with `puppet config print autosign`.
```
Comment (timogoebel) [inline _includes/manuals/nightly/4.3.7_smartproxy_puppetca.md]: ... `can be determined with`

**Review by timogoebel — COMMENTED** [review]
Just one nit. Otherwise: Looks good.

**Comment by ekohl** [comment]
This looks like a very good start, but it's important to also note you need to install the script to do verification. https://github.com/theforeman/smart-proxy/blob/develop/extra/puppet_sign.rb should be deployed somewhere and puppet's autosign setting should point to it. By default it's installed to `/usr/share/foreman-proxy/extra/puppet_sign.rb`. Not sure if permissions need to be fixed as well.

**Comment by dgoetz** [comment]
Thanks for the reviews so far.
@timogoebel: I fixed the wording.
@ekohl: I added an additional section for the Puppet part of the token based mechanism.
So if you both can please re-review.

**Comment by timogoebel** [comment]
@leewaa: Do you mind reviewing as well (especially the English 🇦🇺)?

**Comment by leewaa** [comment]
Sure

**Comment by leewaa** [comment]
LGTM

**Review by timogoebel — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
Looks correct to me

**Comment by tbrisker** [comment]
@timogoebel @leewaa  @ekohl  - is this all valid for both 1.19 and 1.20? or do we need some changes for one or both?

**Comment by dgoetz** [comment]
The token-based provider is only in 1.20, so I would backport the complete text to 1.20 and only the first half to 1.19.

**Comment by dgoetz** [comment]
Backports are done and tests run successfully so feel free to merge from my side.

**Review by ekohl — APPROVED** [review]

**Review by tbrisker — APPROVED** [review]
Thank you very much @dgoetz and all reviewers!

**Comment by ekohl** [comment]
Now that I dive a bit deeper into this: We install the puppet_sign.rb script twice. The one in `/usr/libexec/foreman-proxy/puppet_sign.rb` is executable while the one in `/usr/share` isn't.

**Comment by dgoetz** [comment]
@ekohl: Should I create a follow-up pull-request or do you want to do so?

**Comment by ekohl** [comment]
@dgoetz be my guest. Then I'll continue preparing the 1.21 release :)

**Pull request merged**